### PR TITLE
Remove `IconComponent` and `BadgeComponent` from Navigation Item and update docs

### PR DESCRIPTION
**Pull request by JoanaMMoreira**
- Remove `IconComponent` and `BadgeComponent` props (these can be passed as children)
- Small styling tweaks
- Update stories and storybook docs to match the latest word doc

### CODE REVIEW

**Review by Fercas123 — DISMISSED** [review]

**Review by Fercas123 — DISMISSED** [review]

File: packages/lab/stories/navigation-item/navigation-item.doc.stories.mdx [hunk 1]
```diff
@@ -16,81 +16,113 @@ import HelpAndSupport from "docs/blocks/help-and-support.doc.mdx";
...
-- When content lacks clear categorization or hierarchy.
+- When you need to show different views of content within the same page, consider using [Tabs](?path=/docs/documentation-lab-tabs--page) instead.
+
+- When your content is very specific ,in this case, a minimalistic or single-page design may be more appropriate than a complex navigation structure. Consider using the [Interactable Card](?path=/docs/documentation-core-card--page), or a [Link](?path=/docs/documentation-core-link--page) component within the body of the page.
+
+## Content
+
+- Our recommendation is that all labels should be short and concise.
+
+- Always use labels in sentence case in a Navigation Item.
```
Comment (origami-z) [inline packages/lab/stories/navigation-item/navigation-item.doc.stories.mdx:43]: Does this mean we need a different version for "Vertical group nested" example 

> Label 1 - Level 0

Comment (JoanaMMoreira) [reply packages/lab/stories/navigation-item/navigation-item.doc.stories.mdx:43]: Good spot! I'll update this

File: packages/lab/src/navigation-item/NavigationItem.tsx [hunk 2]
```diff
@@ -21,7 +19,7 @@ export interface NavigationItemProps extends ComponentPropsWithoutRef<"div"> {
    */
   active?: boolean;
   /**
-   * Whether the navigation item has active children.
+   * Whether the parent navigation item has active nested items.
```
Comment (origami-z) [inline packages/lab/src/navigation-item/NavigationItem.tsx:22]: TBH, i don't quite the doc here v.s. the prop `blurSelected`

File: packages/lab/stories/navigation-item/navigation-item.stories.css [hunk 3]
```diff
@@ -8,9 +8,14 @@ ul {
   margin: 0;
 }
 
+.navigationItemContainer {
```
Comment (origami-z) [inline packages/lab/stories/navigation-item/navigation-item.stories.css:11]: Should a layout component be used ?

File: packages/lab/stories/navigation-item/navigation-item.doc.stories.mdx [hunk 4]
```diff
@@ -120,6 +150,22 @@ import { NavigationItem } from "@salt-ds/lab";
...
+| -------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Tab            | <ul><li>Moves focus into the first Navigation item in the group. Focus is set to the entire component.</li><li>If a Navigation item is selected before the Navigation item receives focus, focus is set on the selected item.</li><li>When the Navigation item has nested items, tab again, to move focus on to the next item within the group.</li><li>When Navigation item has focus, move focus out of Navigation item to next component in tab order.</li></ul> |
```
Comment (origami-z) [inline packages/lab/stories/navigation-item/navigation-item.doc.stories.mdx:159]: I don't think this is implemented?
> If a Navigation item is selected before the Navigation item receives focus, focus is set on the selected item

**Review by origami-z — APPROVED** [review]
Non-blocking comments, given we're going to lab

**Review by feliciackh — APPROVED** [review]

**Comment by JoanaMMoreira** [comment]
> Non-blocking comments, given we're going to lab

Thanks! I'll address these in a follow up PR as I have to make another small tweak

File: packages/lab/src/navigation-item/NavigationItem.tsx (hunk 2 above)
Comment (JoanaMMoreira) [reply packages/lab/src/navigation-item/NavigationItem.tsx:22]: I'll review the wording

File: packages/lab/stories/navigation-item/navigation-item.doc.stories.mdx (hunk 4 above)
Comment (JoanaMMoreira) [reply packages/lab/stories/navigation-item/navigation-item.doc.stories.mdx:159]: Yeah we need to rephrase this

**Review by joshwooding — APPROVED** [review]

**Pull request merged**